blog/_posts/2019-02-17-github-situation.markdown

```diff
@@ -0,0 +1,107 @@
+
+To put it simply:
+
+- it's OK to have a lot of big files at the **Releases** section so that you can host stuff;
+- but it's not OK to have big files stored at the repository itself, since Git has poor performance and other issues when the repo gets big.
+
+After [@mfonville] and [@nicholasbuse] had some discussion with the Github support, the result was as follows:
+
+- we shouldn't use these repos to store the APKs the way we did anymore;
+- we should shrink the size of the repos, including the commit history;
+- the fate of the APK repos isn't solved yet (whether they'll be removed or not) - discussion is still open about that;
+- we still can store the OpenGApps builds in the [Releases] section of our repos (not in the APK repos though since it's not yet clear whether they'll be removed or not);
+- the access will be restored for a week to enable us to do maintenance and to solve the problem once and for all.
+
+It was pretty clear that we had to act in order to solve the problem and to restore out infrastructure to continue the builds.
+
+We had multiple options to choose from, including, but not limited to:
+
+- moving to some other public Git hosting, like [GitLab](https://gitlab.com/) or [Bitbucket](https://bitbucket.org/) (this may solve the issue for now, but no one can guarantee that a new hoster wouldn't have a same rule in the future forcing us to move again);
+- hosting the Git repos ourselves (saves us from the ToS nonsense, but leaves us with the hosting problem and the increased costs);
+- restructuring the whole APK repository system, like extracting all the APKs to their own subrepos (this may still happen in the future, but it requires a lot of work which we currently don't have the time for).
+
+### The solution
+
+For now we've decided to pick the second option, as in to self-host our APK repos with [GitLab Community Edition](https://about.gitlab.com/is-it-any-good/).
+
+This gives us three main benefits:
+
+1. It allows to use the same infrastructure for the buildserver, which will only require to redirect the scripts in the [/opengapps] repo to the new Git remotes, hence saving the time;
+2. We can have better control over the content overall;
+3. We can restructure the project later on our terms, with possibly moving away from Github completely and using some filestorage solution to host the releases (e.g. [Amazon AWS S3](https://aws.amazon.com/s3/) / [DigitalOcean Spaces](https://www.digitalocean.com/products/spaces/) to speed up the traffic and ease the load).
+
+Thanks to the hosting which we're now using (btw, big shoutout for [Hetzner](https://www.hetzner.com/), they have really good servers with decent pricing and support), we were able us to accomplish this task without any serious issues.
+
+Here's the current situation:
+
+- APK repos are now hosted at <https://gitlab.nezorfla.me/opengapps> which is a public repository group with full Read access.
```

Comment:
Will read-access not incur high server cost for us?

Reply:
@mfonville Hetzner doesn't have traffic limits, so we should be fine.
The only thing we might need in the future is the server upgrade, but that won't be necessary for a long time.